### ISSUE TITLE
Problem with ARM

### ISSUE THREAD

**Issue by ndebbrecht** [nmslib/hnswlib#202]
Hi everyone!

I have a problem loading the indices. I trained the model on a server architecture and exported the .index files. The model should be loaded on an edge computer running on a armhf architecture (debian buster). When trying to load the .index file I am getting the "RuntimeError: Index seems to be corrupted or unsupported". The file sizes of the index files are all correct and on a unix system I loaded those files without any problems. So it seems to be a problem with the architecture. Is there a possibility to load them without training the model again on the arm environment, which is not as performant?

**Comment by searchivarius** [comment]
@ndebbrecht I guess it's little vs big endian issue. The current index-saving model doesn't create platform-independent representation.

**Comment by ndebbrecht** [comment]
> @ndebbrecht I guess it's little vs big endian issue. The current index-saving model doesn't create platform-independent representation.

So there is no other possibility than recreating them on a ARM platform to use them on a ARM platform?

**Comment by searchivarius** [comment]
@ndebbrecht it's sure possible to change the code :-)

**Comment by yurymalkov** [comment]
@ndebbrecht 
Yes, as @searchivarius suggested, it would require some changes in the code. 

I see two options:
1) directly change little/big endian in loading index. Not sure if floats also need to change
2) make ascii-based serialization (a more general solution).
Both of the approaches would require to add custom serialization of the data.

I can help with that (a little) if you like.

**Comment by searchivarius** [comment]
Floats sure need to change. You have 4 float bytes stored as x y z t on Intel, but it will be loaded something like t z y x on ARM. There are posix functions to deal with it. Alternatively, there's a boost library. But it's some non-trivial amount of work to save/load in endianness-neutral format.

**Comment by yurymalkov** [comment]
@searchivarius 
I see. Then, ascii probably would be a better solution.

**Comment by WellWang-S** [comment]
> @searchivarius I see. Then, ascii probably would be a better solution.

same problem.How did you solve it?
